portal/worker/src/walrus-sites-sw.ts

```diff
@@ -50,8 +54,40 @@ self.addEventListener("fetch", async (event) => {
     console.log("Parsed URL: ", parsedUrl);
 
     if (requestDomain == portalDomain && parsedUrl && parsedUrl.subdomain) {
-        console.log("fetching from the service worker");
-        event.respondWith(resolveAndFetchPage(parsedUrl));
+        event.respondWith((async () => {
+            // Clean the cache of expired entries
+            await cleanExpiredCache();
+            if (!('caches' in self)) {
+                // When not being in a secure context, the Cache API is not available.
+                console.warn('Cache API not available');
+                return await resolveAndFetchPage(parsedUrl);
+            }
+
+            const cache = await caches.open(cacheName);
+            const cachedResponse = await cache.match(urlString);
+            let isCacheSameAsNetwork: boolean;
+            try {
+                if (cachedResponse) {
+                    isCacheSameAsNetwork = await checkCachedVersionMatchesOnChain(cachedResponse)
+                }
+            } catch (e) {
+                console.error("Error checking cache version against chain:", e);
+            }
+            if (cachedResponse && isCacheSameAsNetwork) {
+                console.log("Cache hit!", urlString);
+                return cachedResponse;
+            } else {
+                console.log("Cache miss!", urlString);
+                const resolvedPage = await resolveAndFetchPage(parsedUrl);
+
+                // The urlString (i.e. the cache key) will also contain the timestamp
+                // to ensure that the cache is not stale.
+                // The delimiter used is an illegal URI character
+                // to avoid any conflicts with the actual URL.
```

Comment:
Question: where is this timestamp set in the url? Is this an outdated comment?

Reply:
Good catch, that's a remnant of the first implementation. I will remove it ✍️